src/algebra/parity.lean

```diff
@@ -84,15 +87,50 @@ lemma even.neg_one_pow (h : even n) : (-1 : α) ^ n = 1 := by rw [h.neg_pow, one
 
 end monoid
 
+section comm_monoid
+variable [comm_monoid α]
+
 @[to_additive]
-lemma is_square.mul_is_square [comm_monoid α] {m n : α} (hm : is_square m) (hn : is_square n) :
+lemma is_square.mul_is_square {m n : α} (hm : is_square m) (hn : is_square n) :
   is_square (m * n) :=
 begin
   rcases hm with ⟨m, rfl⟩,
   rcases hn with ⟨n, rfl⟩,
   refine ⟨m * n, mul_mul_mul_comm m m n n⟩,
 end
 
+-- @[simp]
+-- TODO (khw): This an the following commented out @[simp]s appear because they
+-- _should_ be simp lemmas but break the simp engine when included. Thus, we
+-- comment them out for a later day
```

Comment:
I'm not convinced that they should be `simp` lemmas. I think that simp is unlikely to find the preconditions, so I would leave them out of the default simp set.

Reply:
Ah that might explain things. This was in comparison to the analogous `squarefree` lemmas, which are `simp`. Happy to leave them out

Reply:
OK, removed